### PR TITLE
Manage organization-mgt.xml & organization-mgt.xml.j2 files

### PR DESCRIPTION
**Pull request by sadilchamishka**
```
<file>
    <source>
        ../p2-profile-gen/target/wso2carbon-core-${carbon.kernel.version}/repository/conf/organization-mgt.xml
    </source>
    <outputDirectory>wso2is-${pom.version}/repository/conf</outputDirectory>
    <fileMode>644</fileMode>
 </file>
```

Ex - https://github.com/wso2/product-is/pull/16341/files#diff-827afa46b6d3fb2fb1210e459a00f4702c81429135daffd5205f198d0bf0bab4R1030-R1043

### CODE REVIEW

File: features/org.wso2.carbon.identity.organization.management.core.feature/resources/organization-mgt.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<!--
+  ~ Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```
Comment (inthirakumaaran) [inline features/org.wso2.carbon.identity.organization.management.core.feature/resources/organization-mgt.xml:2]: Don't we need the `All Rights Reserved.` part? check the other places

Comment (AnuradhaSK) [reply features/org.wso2.carbon.identity.organization.management.core.feature/resources/organization-mgt.xml:2]: It's not required for public source code license

File: features/org.wso2.carbon.identity.organization.management.core.feature/resources/organization-mgt.xml.j2 [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<!--
+~ Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```
Comment (AnuradhaSK) [inline features/org.wso2.carbon.identity.organization.management.core.feature/resources/organization-mgt.xml.j2:2]: check the license header

Comment (sadilchamishka) [reply features/org.wso2.carbon.identity.organization.management.core.feature/resources/organization-mgt.xml.j2]: Sorry, missed that :|  Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/74/commits/0646e20ade6dd091350510081218918da4a1d5d0

File: features/org.wso2.carbon.identity.organization.management.core.feature/pom.xml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+    <parent>
+        <groupId>org.wso2.carbon.identity.organization.management.core</groupId>
```
Comment (AnuradhaSK) [inline features/org.wso2.carbon.identity.organization.management.core.feature/pom.xml:20]: didn't we have a carbon feature for org.wso2.carbon.identity.organization.management.core?
If not, how it was packed to the IS by default as a feature earlier?

Comment (sadilchamishka) [reply features/org.wso2.carbon.identity.organization.management.core.feature/pom.xml:20]: Yes we have. But in order to bundle these resource files, another feature module was added. It was the convention in both OAuth repo and framework repo.

Screenshot 2023-08-03 at 11 43 03

Also, with this effort, previous feature module was refactored from
`org.wso2.carbon.identity.organization.management.core.server.feature` to `org.wso2.carbon.identity.organization.management.core.feature` in order to comply with the convensions used in the other repos.

Comment (AnuradhaSK) [reply features/org.wso2.carbon.identity.organization.management.core.feature/pom.xml:20]: I also checked the code bases. we can just go with `org.wso2.carbon.identity.organization.management.core.server.feature` including the service + files
In other components, we have a separate feature to the backend component, a carbon UI-related component and a composite feature.
`<component name>.server.feature` is the feature name used to pack backend service + config files
Refer: https://wso2.com/library/tutorials/2014/03/how-to-create-a-feature-from-wso2-carbon-component/

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagementInitializeImpl.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Organization Management feature enabling initialization implementation.
+ */
+public class OrganizationManagementInitializeImpl implements OrganizationManagementInitialize {
+
+    @Override
+    public boolean isOrganizationManagementEnabled() {
+
+        return true;
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagementInitializeImpl.java:29]: This should be improved to return true/false based on a config. (Enable or disable B2B feature for the product)

File: features/org.wso2.carbon.identity.organization.management.core.feature/pom.xml (hunk 3 above)
Comment (sadilchamishka) [reply features/org.wso2.carbon.identity.organization.management.core.feature/pom.xml]: Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/74/commits/d83a5778490e205d969d0ffeec461db8f0e288f4

File: components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagementInitializeImpl.java (hunk 4 above)
Comment (inthirakumaaran) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagementInitializeImpl.java:29]: What happened to run B2B by using different profiles at startup? AFAIR there is a discussion on that angle

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagementInitializeImpl.java]: Yes. due to that complexities, removed this implementation and decided to have it at the end of onboarding the required resources.

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**